src/algorithms/reco/FarForwardNeutronReconstruction.cc

```diff
@@ -41,9 +46,11 @@ namespace eicrecon {
       }
       if (Etot>0){
           auto rec_part = out_neutrons->create();
+          double corr=calc_corr(Etot);
+          Etot=Etot/(1+corr);
```

Comment:
Instead of adding ad-hoc functionality to what is essentially ReconstructedNeutralParticles factory, can we add evaluator expression support to ClusterRecoCoG sampFrac parameter, like it's done in #1410.

Reply:
That would need to make sampFrac depend not only on cellID, but also on original cluster energy.

Reply:
A single neutron may produce multiple topological clusters, so the correction for single-neutron events would be different depending on the total shower energy, rather than having separate sampling fractions for each cluster.  Further, the sampling fractions we are using are based on an EM response (since the ZDC Hcal will measure not only hadrons but also mid-to-high energy photons as well).  This correction compensates for the different between EM and hadronic responses.  So, we'd prefer to keep this as we have implemented it in this pull request.